### ISSUE TITLE
Device Architecture

### ISSUE THREAD

**Issue by jrahma** [xamarin/Essentials#863]
Hi,

Please add an option to get device architecture, e.g. x86, armeabi-v7a, etc

Thanks,
[Jassim](https://www.jassimrahma.net)

**Comment by dotMorten** [comment]
@jrahma I was looking at the PR, and I was a little bit confused about what the PR is trying to attempt, and it seems to do different things on different platforms.
I think if there was some solid use-cases for why this is needed, it would be easier to design the API specifically around that.

Could you clarify why you need to get the device architecture? What sort of experience or app are you trying to build where this is required, and why?

**Comment by jrahma** [comment]
@dotMorten I am trying to develop a device information app using Xamarin Forms

**Comment by dotMorten** [comment]
Just to play devils advocate then: I you are planning on creating an app that shows very device specific information, would it be reasonable to expect those APIs to be available cross-platform, as many many things would be very specific to the individual platform?

Also what sort of information are you looking for in this specific case: Whether the app is 32 or 64 bit, whether the device is 32 or 64 bit, and what sort of processor architecture it is etc etc.

**Comment by jamesmontemagno** [comment]
After reviewing with the team, we have decided that there isn't a good cross-platform API to understand "supported" vs cpu architecture that is consistent across all of the platforms. We also analyzed the feature and believe that the use case for it is pretty low and the code would be easy enough for a developer to implement if they need this specific functionality. For those reasons we are going to close it out.
